lib/hystrix-stats.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const stream = require('stream');
+
+const hystrixFormatter = require('./hystrix-formatter');
+/*
+  This class needs a function that outputs a stream of circuit data
+
+  Will also need to be a listener on status events
+*/
+
+class HystrixStats {
+  constructor () {
+    this._readableStream = new stream.Readable({
+      objectMode: true
+    });
+
+    // Need a _read() function to satisfy the protocol
+    this._readableStream._read = () => {};
+    this._readableStream.resume();
+
+    this._hystrixStream = new stream.Transform({
+      objectMode: true
+    });
+
+    // Need a _transform function to satisfy the protocol
+    this._hystrixStream._transform = this._hystrixTransformer;
+    this._hystrixStream.resume();
+
+    this._readableStream.pipe(this._hystrixStream);
+  }
+
+  // The stats coming in should be already "Reduced" and in
+  _hystrixTransformer (stats, encoding, cb) {
+    console.log('transformer', stats, encoding);
+
+    const formattedStats = hystrixFormatter(stats);
+
+    // Need to take the stats and map them to the hystrix format
+    return cb(null, `data: ${JSON.stringify(formattedStats)}\n\n`);
+  }
+
+  // Just a regular public function to get a handle on the stream
+  getHystrixStream () {
+    return this._hystrixStream;
+  }
+
+  // This will take the stats data from the listener and push it on the stream to be transformed
+  _hystrixSnapshotListener (stats) {
+    console.log('_hystrixSnapshotListener', stats);
+    this._readableStream.push(stats);
+  }
+
+  // Register the instance of a circuit to listen for a snapshot event
+  // during that event, call the hystrixSnapshotListener to do the stream stuff
+  register (circuitInstance) {
+    circuitInstance.on('snapshot', this._hystrixSnapshotListener.bind(this));
```

Comment:
Currently it's the `Status` object that emits `snapshot`. That could change if you think it's better suited in `CircuitBreaker`, but I think it would be unnecessary indirection. It would mean that the circuit breaker listens to the status object for snapshot events and just propagates them to the client.